static/js/stream_edit.js

```diff
@@ -478,6 +506,96 @@ export function initialize() {
+
+        stream_data.clean_up_description(stream);
+        const sub = stream_settings_data.get_sub_for_settings(stream);
+
+        const all_settings = stream_settings(sub);
+        const other_settings = [];
+        const notification_settings = all_settings.filter((setting) => {
+            if (setting.is_notification_setting) {
+                return true;
+            }
+            other_settings.push(setting);
+            return false;
+        });
+
+        let value = "retain_for_period";
+        if (stream.message_retention_days === null) {
+            value = "realm_default";
+        } else if (stream.message_retention_days === settings_config.retain_message_forever) {
+            value = "unlimited";
+        }
+
+        const change_general_settings = render_change_general_stream_settings({
+            sub,
+            notification_settings,
+            other_settings,
+            message_retention_text: get_retention_policy_text_for_subscription_type(sub),
+            stream_privacy_policy_values: stream_data.stream_privacy_policy_values,
+            stream_privacy_policy: stream_data.get_stream_privacy_policy(stream_id),
+            stream_post_policy_values: stream_data.stream_post_policy_values,
+            is_owner: page_params.is_owner,
+            disable_message_retention_setting:
+                !page_params.zulip_plan_is_not_limited || !page_params.is_owner,
+            stream_message_retention_days: value,
+            retention_days: stream.message_retention_days,
+            org_level_message_retention_setting:
+                get_display_text_for_realm_message_retention_setting(),
+            zulip_plan_is_not_limited: page_params.zulip_plan_is_not_limited,
+            upgrade_text_for_wide_organization_logo:
+                page_params.upgrade_text_for_wide_organization_logo,
+            is_stream_edit: true,
+            max_stream_name_length: page_params.max_stream_name_length,
+            max_stream_description_length: page_params.max_stream_description_length,
+        });
+
+        $(".changeable-settings").replaceWith(change_general_settings);
+        change_stream_message_retention_days_block_display_property(value);
+    });
+
+    $("#manage_streams_container").on("click", "#save-settings", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const stream_id = get_stream_id(e.target);
+        const stream = sub_store.get(stream_id);
+        stream_data.clean_up_description(stream);
+        const sub = stream_settings_data.get_sub_for_settings(stream);
+        change_stream_settings(e, sub);
+
+        const show_general_settings = render_show_general_stream_settings({
+            sub,
+            stream_post_policy_values: stream_data.stream_post_policy_values,
+            message_retention_text: get_retention_policy_text_for_subscription_type(sub),
+        });
+
+        $(".changeable-settings").replaceWith(show_general_settings);
```

Comment:
This should be done, only if the update succeeds from API. If not then we should probably show the error while displaying edit elements only.

Reply:
okay, how do i do this? 
So how do i find out if the update succeeded?
And how do i show the error?
And what do you mean with "display edit elements only"?

Reply:
You can use `success_continuation` and `error_continuation` parameters of `settings_ui.do_settings_change`. I meant that if the update does not succeeds we should show the error message while showing the edit input elements.